src/extension.ts

```diff
@@ -96,7 +96,20 @@ export async function activate(context: vscode.ExtensionContext) {
               return item;
             })
             return new vscode.CompletionList(items, /*isIncomplete=*/ true);
-          }
+          },
+      // VSCode applies fuzzy match only on the symbol name, thus it throws away
+      // all results if query token is a prefix qualified name.
+      // By adding the containerName to the symbol name, it prevents VSCode from
+      // filtering out any results, e.g. enable workspaceSymbols for qualified
+      // symbols.
+      provideWorkspaceSymbols: async (query, token, next) => {
+        let symbols = await next(query, token);
+        return symbols.map(symbol => {
+          if (symbol.containerName)
```

Comment:
This results in containerName being displayed twice (once as part of Name and once separately).

![image](https://user-images.githubusercontent.com/548993/82269929-68fe3380-9973-11ea-8579-f7406a44d077.png)

I'd suggest:

 - only do this if there's a container name *and* the query contains ::
 - clear out containerName when you do this
 - leave a comment explaining that we don't want to display containerName twice

This isn't so bad as the aligned blue in the LHS will still make it easy to visually scan the rest of the symbol names.

Reply:
> This results in containerName being displayed twice (once as part of Name and once separately).

this is a good point. It is not good to displayed the container name twice. I find the `symbol.containerName` being displayed in a less discoverable way (compare with the `symbol.Name`), what do you think that we always clean the container name, and use the `qualified name` as the `SymbolName`

Reply:
Yeah actually I think you're right, this looks better, since the name gets highlighted anyway and now the namespace doesn't run together into the filename.